sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormTrainingAsyncClient.java

```diff
@@ -0,0 +1,304 @@
+     * also need to be included when searching for content to be preprocessed.
+     * @param filePrefix A case-sensitive prefix string to filter documents in the source path
+     * for training. For example, when using a Azure storage blob Uri, use the prefix to restrict
+     * sub folders for training.
+     * @param pollInterval Duration between each poll for the operation status. If none is specified, a default of
+     * 5 seconds is used.
+     *
+     * @return A {@link PollerFlux} that polls the extract receipt operation until it
+     * has completed, has failed, or has been cancelled.
+     */
+    public PollerFlux<OperationResult, CustomFormModel> beginTraining(String fileSourceUrl,
+        boolean useLabelFile, boolean includeSubFolders, String filePrefix, Duration pollInterval) {
+        Objects.requireNonNull(fileSourceUrl, "'fileSourceUrl' cannot be null.");
+        Objects.requireNonNull(useLabelFile, "'useLabelFile' cannot be null.");
+        final Duration interval = pollInterval != null ? pollInterval : Duration.ofSeconds(5);
+
+        return new PollerFlux<OperationResult, CustomFormModel>(
+            interval,
+            getTrainingActivationOperation(fileSourceUrl, includeSubFolders, filePrefix, useLabelFile),
+            createTrainingPollOperation(),
+            (activationResponse, context) -> Mono.error(new RuntimeException("Cancellation is not supported")),
+            fetchTrainingModelResultOperation());
+    }
+
+    /**
+     * Get detailed information for a specified custom model id.
+     *
+     * @param modelId Model identifier.
+     *
+     * @return The detailed information for the specified model.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<CustomFormModel> getCustomModel(String modelId) {
+        try {
+            return getCustomFormModelWithResponse(modelId).flatMap(FluxUtil::toMono);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Get detailed information for a specified custom model id with Http response
+     *
+     * @param modelId Model identifier.
+     *
+     * @return A {@link Response} containing the requested {@link CustomFormModel model}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<CustomFormModel>> getCustomFormModelWithResponse(String modelId) {
+        try {
+            return withContext(context -> getCustomFormModelWithResponse(modelId, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    Mono<Response<CustomFormModel>> getCustomFormModelWithResponse(String modelId, Context context) {
+        Objects.requireNonNull(modelId, "'modelId' cannot be null");
+        return service.getCustomModelWithResponseAsync(UUID.fromString(modelId), context, true)
```

Comment:
If the model ID is a `UUID` should we accept that in the API instead?

Reply:
Expecting a string modelID is more approachable for the user so decided to go with it.